### ISSUE TITLE
py-polars sdist not buildable since 0.19.4

### ISSUE THREAD

**Issue by 0xbe7a** [pola-rs/polars#11371]
When attempting to run `maturin build` inside the source tarball generated by `maturin sdist` it fails with 

```
error: no matching package named `arrow` found
location searched: /private/tmp/source_code/polars-0.19.5/local_dependencies/arrow
required by package `polars-core v0.33.2 (/private/tmp/source_code/polars-0.19.5/local_dependencies/polars-core)`
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error:
```

Steps to reproduce:
1. Run `maturin sdist` inside py-polars
2. Extract the source tarball somewhere
3. Run `maturin build` inside the extracted tarball

Git bisected yielded https://github.com/pola-rs/polars/commit/122b2ed4ce9757541c2d07ee8cfd617d6be5e4c3 as the breaking commit.

Curiously the tarballs that got uploaded to PyPy also fail since 0.19.4, but with a different error:

```
error: failed to load manifest for dependency `polars`

Caused by:
  failed to parse manifest at `/Users/belastoyan/Downloads/polars-0.19.5/local_dependencies/polars/Cargo.toml`

Caused by:
  error inheriting `polars-algo` from workspace root manifest's `workspace.dependencies.polars-algo`

Caused by:
  `workspace.dependencies` was not defined
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error:
  ```

**Comment by ritchie46** [comment]
Just curious. Does 0.19.5 work? There were some reverts.

**Comment by 0xbe7a** [comment]
No

```
➜  polars-0.19.5 head Cargo.toml 
[package]
name = "py-polars"
version = "0.19.5"
edition = "2021"

[lib]
name = "polars"
crate-type = ["cdylib"]

[dependencies]
➜  polars-0.19.5 maturin build
error: failed to load manifest for dependency `polars`

Caused by:
  failed to parse manifest at `/Users/belastoyan/Downloads/polars-0.19.5/local_dependencies/polars/Cargo.toml`

Caused by:
  error inheriting `polars-algo` from workspace root manifest's `workspace.dependencies.polars-algo`

Caused by:
  `workspace.dependencies` was not defined
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error: 
➜  polars-0.19.5 
```

**Comment by 0xbe7a** [comment]
This is a maturin bug, i was able to create a minimal example resulting in the same behaviour: https://github.com/PyO3/maturin/issues/1781

**Comment by chebee7i** [comment]
Any workaround for now? I'm trying to build a `bigidx` variant of polars for conda.

Aside: Do you have a plan for making the conda-forge feedstock work with more customization?

**Comment by chebee7i** [comment]
Related to this...I tried to build 0.19.3 with `bigidx`: https://github.com/conda-forge/polars-feedstock/issues/178

**Comment by stinodego** [comment]
A new maturin version was released with the fix. I have tested and the `sdist` is now installable. The next Polars release will ship with a proper sdist :+1: 

> Any workaround for now?

Use the `0.19.3` version until we release the new version, or clone the repo and build the sdist yourself using the latest maturin version.

> Aside: Do you have a plan for making the conda-forge feedstock work with more customization?

I'm not familiar with how conda-forge works, but I suppose it would be possible to add a separate feedstock for `polars-u64-idx` and `polars-lts-cpu`. Our team is not planning to set those up, but I suppose someone else could do that?
